src/modules/CRUD/views/MapPlayground/components/LayersControl/LayersControl.js

```diff
@@ -13,24 +14,74 @@ export class LayersControl extends AbstractMapControl {
   static propTypes = {
     layers: PropTypes.arrayOf(
       PropTypes.shape({}),
-    ).isRequired,
+    ),
+    relations: PropTypes.arrayOf(
+      PropTypes.shape({}),
+    ),
+    getMapStyle: PropTypes.func,
   }
 
   static defaultProps = {
+    layers: [],
+    relations: [],
+    getMapStyle () {},
     onChange () {},
   }
 
   componentDidMount () {
-    const { layers, map } = this.props;
+    const { layers = [], map } = this.props;
     layers.forEach(({ id }) => {
       this.setLayoutProperty(id, map.getLayoutProperty(id, 'visibility') === 'visible');
     });
   }
 
+  componentWillUnmount () {
+    const { map, relations } = this.props;
+    relations.forEach(({ crud_view_pk: id }) => {
+      const layerId = this.getRelationLayerID(id);
+      if (map.getSource(layerId)) {
+        map.setLayoutProperty(layerId, 'visibility', 'none');
+      }
+    });
+  }
+
   onChange = ({ target: { value, checked } }) => {
     this.setLayoutProperty(value, checked);
   }
 
+  getRelationLayerID = id => {
+    const { featureID } = this.props;
+    return `CRUD-${featureID}-relation-${id}`;
+  }
+
+  onChangeRelation = async ({ target: { value, checked } }, geojson) => {
+    const { map, getMapStyle, layers } = this.props;
+    const currentSourceAndLayerID = this.getRelationLayerID(value);
+    if (checked) {
```

Comment:
May be an early return can enhance readability here ?

Reply:
Done